llama_stack/apis/post_training/post_training.py

```diff
@@ -111,11 +116,7 @@ class PostTrainingSFTRequest(BaseModel):
     validation_dataset_id: str
 
     algorithm: FinetuningAlgorithm
```

Comment:
we should kill this completely and just use the tagged union I suggested below

Reply:
we can kill this. But given full finetune won't have its algorithm_config, shall we do sth like 
if algorithm_config is None:  full finetune recipe
else if isinstance(algorithm_config, LoraFinetuningConfig): lora finetune recipe
else if isinstance(algorithm_config, QATFinetuningConfig): qat finetune recipe